lib/firestore/delete.js

```diff
@@ -232,6 +256,22 @@ FirestoreDelete.prototype._getDescendantBatch = function(allDescendants, batchSi
     });
 };
 
+/**
+ * Delete a single Firestore document.
+ *
+ * For document format see:
+ * https://firebase.google.com/docs/firestore/reference/rest/v1beta1/Document
+ *
+ * @param {object} doc a Document object to delete.
+ * @return {Promise} a promise for the delete operation.
+ */
+FirestoreDelete.prototype._deleteDocument = function(doc) {
+  return api.request("DELETE", "/v1beta1/" + doc.name, {
```

Comment:
could you refactor all the Firestore API calls and put them in a new file lib/gcp/firestore.js? (Similiar to lib/gcp/cloudfunctions.js)

Reply:
I moved the generally useful API calls.  The deep descendant queries I left here because they use fields from the object and they are not generally useful.